### PR TITLE
Add missing OL releases to RID graph

### PR DESCRIPTION
**Pull request by Mr-Tao**
In PR, I’ve tried to address comments that #66783 received.

I’m very much in favour of adding only major versions to the RID graph, as @omajid suggested in https://github.com/dotnet/runtime/pull/66783#issuecomment-1090511904. In Oracle Linux, we have been adding minor versions for Dotnet versions up to 7.9 and 8.7 so far.

Since all minor versions should be compatible, dropping all minor versions from the graph may be possible. But I see that this didn’t happen with rhel minor versions already present in the graph, which lets me think about whether there are potential repercussions from dropping them. 🤔 I don’t know for sure, so I have kept minor versions.

> But since we are now adding OL, we need the parent RHEL versions (is this a strict requirement? Can anyone confirm) https://github.com/dotnet/runtime/pull/66783#discussion_r911104192

I tried generating the graph without adding respective rhel RIDs. The task failed as expected, complaining about missing parents.

### CODE REVIEW

**Comment by omajid** [comment]
I am happy there are no minor-version RIDs for OL 9 :tada: 

> Since all minor versions should be compatible, dropping all minor versions from the graph may be possible. But I see that this didn’t happen with rhel minor versions already present in the graph, which lets me think about whether there are potential repercussions from dropping them. thinking I don’t know for sure, so I have kept minor versions.

I think it's a compatibility concern. If a user has used the RID in the past and we drop the RID in the graph, that might break applications, or at prevent the build/publish commands that were previously working from continuing to work.

I don't know if these concerns still apply when adding it to repo here, because those RIDs weren't present previously so they wouldn't be treated as "removed"/"breaking" if they just not added.

So maybe we can leave out the minor version RIDs from this PR? Will this really impact customers given the RID graph on nuget.org didn't have these versions already?

Anyway, I wont block the addition of RHEL minor version IDs here. I don't see a harm.

**Comment by agocke** [comment]
While we're contemplating deprecating the whole RID graph (https://github.com/dotnet/runtime/issues/83246), I'm going to close this for now. We can re-open if we end up not going in that direction

**Comment by ViktorHofer** [comment]
@richlander until we execute on the RID plan changes, should we keep merging these PRs or close them?

**Comment by tmds** [comment]
> We are already following Red Hat by not introducing any more RIDs for minor versions of Oracle Linux 9.

You can't omit these minor versions until you update .NET to ignore the minor.

That means adding code here:

https://github.com/dotnet/runtime/blob/31e4f404c218eae7ba999c4df5346d30f971451c/src/native/corehost/hostmisc/pal.unix.cpp#L766-L770

(and similar for .NET 6).

**Comment by ViktorHofer** [comment]
ping @richlander regarding my above question

**Comment by carlossanlop** [comment]
Reopening this PR to continue the conversation. I closed the release/6.0 PR version of this change, since we would first need to merge the main version before taking a servicing version: https://github.com/dotnet/runtime/pull/83681

Ping @richlander

**Comment by Mr-Tao** [comment]
> You can't omit these minor versions until you update .NET to ignore the minor.
> 
> That means adding code here:
> 
> https://github.com/dotnet/runtime/blob/31e4f404c218eae7ba999c4df5346d30f971451c/src/native/corehost/hostmisc/pal.unix.cpp#L766-L770
> 
> (and similar for .NET 6).

Thanks for pointing this out, @tmds. Do you think that there’s anything else that needs to be added besides this change? https://github.com/dotnet/runtime/pull/81115/files#diff-b2bd8511e88327d8f146bc460e79182cc6826e2b90039dffc4f5f89c5467d12a

File: src/libraries/Microsoft.NETCore.Platforms/src/runtimeGroups.props [hunk 1]
```diff
@@ -225,12 +230,12 @@
...
       <Parent>linux</Parent>
       <Architectures>x64</Architectures>
-      <Versions>7;7.0;7.1;7.2;7.3;7.4;7.5;7.6</Versions>
+      <Versions>7;7.0;7.1;7.2;7.3;7.4;7.5;7.6;7.7;7.8;7.9</Versions>
     </RuntimeGroup>
     <RuntimeGroup Include="rhel">
       <Parent>linux</Parent>
       <Architectures>x64;arm64</Architectures>
-      <Versions>8;8.0;8.1</Versions>
+      <Versions>8;8.0;8.1;8.2;8.3;8.4;8.5;8.6;8.7</Versions>
```
[the remark this answers is not in the document]
Comment (Mr-Tao) [reply src/libraries/Microsoft.NETCore.Platforms/src/runtimeGroups.props:238]: I _suppose_ that dotnet would build fine. If this is safe, why does RedHat still keep minors for 8? Could this be an issue for customer applications? :thinking:

[the remark this answers is not in the document]
Comment (omajid) [reply src/libraries/Microsoft.NETCore.Platforms/src/runtimeGroups.props:238]: > If this is safe, why does Red Hat still keep minors for 8

It's mostly due to backwards compatibility. We added them to the RID graph for 8 before we finished all the fixes (eg, the pal.unix.cpp stuff), and now we can't remove them.

No one should be depending on the minor versions. It makes everyone's life harder.

**Comment by MichaelSimons** [comment]
@agocke, @carlossanlop, @richlander - What is the plan for requests like this to update the RID graph?  I know there is the  [proposed RID plan](https://github.com/dotnet/runtime/issues/83246).  What is the timeline for implementing it?  This issue is affecting folks wanting to source-build .NET.

**Comment by agocke** [comment]
RID changes are in and breaking change is filed: https://learn.microsoft.com/en-us/dotnet/core/compatibility/deployment/8.0/rid-asset-list

RID graph is now frozen.

**Pull request closed without merging**